src/util/kb_mark_up.py

```diff
@@ -0,0 +1,37 @@
+from telegram import InlineKeyboardButton, InlineKeyboardMarkup
+
+
+def settings_kb():
+    button_list = [
+        InlineKeyboardButton("Toggle Menu Visibility", callback_data="settings.hidden"),
+        InlineKeyboardButton("View Favourite Foods", callback_data="settings.favorite"),
+        InlineKeyboardButton("View Notification Settings", callback_data="settings.notification"),
+        InlineKeyboardButton("Back to start", callback_data="start.home")
+    ]
+    return InlineKeyboardMarkup(build_menu(button_list, n_cols=1))
+
+
+def hidden_cuisine_kb(hidden_cuisine):
+    cuisines = [
+        "self_service", "western", "dim_sum_congee_noodle", "asian",
+        "asian_vegetarian", "malay", "halal_vegetarian", "grab_and_go",
+        "noodle", "vegetarian", "soup", "indian",
+    ]
+    kb_markup = list(
+        map(lambda c:
+            InlineKeyboardButton(('❌ ' if c in hidden_cuisine else '✅ ') + c, callback_data="menu." + c),
+            cuisines))
+    setting_button = InlineKeyboardButton("settings", callback_data="settings.home")
+    return InlineKeyboardMarkup(build_menu(kb_markup, n_cols=2, footer_buttons=setting_button))
+
+
+def build_menu(buttons,
+               n_cols,
+               header_buttons=None,
```

Comment:
actually if you're putting an s here, then this should be an array of buttons not just a single button right?

Reply:
yup.

Reply:
So if I pass in a list of 2 buttons now, it will insert 2 rows of 1 button each, or 1 row of 2 buttons? The syntax here a bit confusing, cuz you're inserting and appending [header_buttons] instead of header_buttons